Cabal/Distribution/Utils/LogProgress.hs

```diff
@@ -15,21 +17,40 @@ import Distribution.Utils.Progress
 import Distribution.Verbosity
 import Distribution.Simple.Utils
 import Text.PrettyPrint (Doc, (<+>), text, render)
-import Control.Monad (when)
+
+data LogEnv = LogEnv {
+        le_verbosity :: Verbosity,
+        le_context   :: [Doc]
+    }
 
 -- | The 'Progress' monad with specialized logging and
 -- error messages.
-newtype LogProgress a = LogProgress (Progress LogMsg Doc a)
-    deriving (Functor, Applicative, Monad)
+newtype LogProgress a = LogProgress { unLogProgress :: LogEnv -> Progress LogMsg Doc a }
```

Comment:
Not a big deal, but isn't there a Reader(T) somewhere from a library that `Cabal` already depends on?